File: src/main/java/pl/yalgrin/playnite/simplesync/repository/CategoryRepository.java
package pl.yalgrin.playnite.simplesync.repository;

import org.springframework.data.r2dbc.repository.Query;
import org.springframework.stereotype.Repository;
import pl.yalgrin.playnite.simplesync.domain.Category;
import reactor.core.publisher.Flux;

@Repository
public interface CategoryRepository extends ObjectRepository<Category> {
    @Query("select c.id from playnite_category c order by c.id asc")
    Flux<Long> findAllIds();
}

File: src/main/kotlin/pl/yalgrin/playnite/simplesync/library/repository/CategoryRepository.kt
package pl.yalgrin.playnite.simplesync.library.repository

import org.springframework.data.r2dbc.repository.Query
import org.springframework.stereotype.Repository
import pl.yalgrin.playnite.simplesync.library.domain.Category
import reactor.core.publisher.Flux

@Repository
interface CategoryRepository : ObjectRepository<Category> {
    @Query("select c.id from playnite_category c order by c.id")
    override fun findAllIds(): Flux<Long>
}
